ndsl/comm/__init__.py

```diff
@@ -0,0 +1,15 @@
+from .boundary import SimpleBoundary
+from .caching_comm import (
+    CachingCommData,
+    CachingCommReader,
+    CachingCommWriter,
+    CachingRequestReader,
+    CachingRequestWriter,
+    NullRequest,
+)
+from .comm_abc import Comm, Request
+from .communicator import CubedSphereCommunicator, TileCommunicator
+from .local_comm import AsyncResult, ConcurrencyError, LocalComm
+from .mpi import MPIComm
+from .null_comm import NullAsyncResult, NullComm
+from .partitioner import CubedSpherePartitioner, TilePartitioner
```

Comment:
Do we also want to expose the Partitioner and Communicator abstract base classes here for typing purposes?